cli/command/environment_command_test.go

```diff
@@ -193,3 +194,115 @@ func TestEnvironmentSetMinCount_userInputErrors(t *testing.T) {
 		}
 	}
 }
+
+func TestEnvironmentLink(t *testing.T) {
+	tc, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	command := NewEnvironmentCommand(tc.Command())
+
+	tc.Resolver.EXPECT().
+		Resolve("environment", "name1").
+		Return([]string{"id1"}, nil)
+
+	tc.Resolver.EXPECT().
+		Resolve("environment", "name2").
+		Return([]string{"id2"}, nil)
+
+	tc.Client.EXPECT().
+		CreateLink("id1", "id2").
+		Return(nil)
+
+	c := getCLIContext(t, Args{"name1", "name2"}, nil)
+	if err := command.Link(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestEnvironmentLink_userInputErrors(t *testing.T) {
+	tc, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	command := NewEnvironmentCommand(tc.Command())
+
+	contexts := map[string]*cli.Context{
+		"Missing SOURCE arg":      getCLIContext(t, Args{}, nil),
+		"Missing DESTINATION arg":     getCLIContext(t, Args{"name"}, nil),
+	}
+
+	for name, c := range contexts {
+		if err := command.Link(c); err == nil {
+			t.Fatalf("%s: error was nil!", name)
+		}
+	}
+}
+
+func TestEnvironmentLink_duplicateEnvironmentID(t *testing.T) {
+	tc, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	command := NewEnvironmentCommand(tc.Command())
+
+	tc.Resolver.EXPECT().
+		Resolve("environment", gomock.Any()).
+		Return([]string{"id1"}, nil).
+		Times(2)
+
+	c := getCLIContext(t, Args{"name1", "name2"}, nil)
+	if err := command.Link(c); err == nil {
+		t.Fatal("error was nil!")
+	}
+}
+
+func TestEnvironmentUnink(t *testing.T) {
+	tc, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	command := NewEnvironmentCommand(tc.Command())
+
+	tc.Resolver.EXPECT().
+		Resolve("environment", "name1").
+		Return([]string{"id1"}, nil)
+
+	tc.Resolver.EXPECT().
+		Resolve("environment", "name2").
+		Return([]string{"id2"}, nil)
+
+	tc.Client.EXPECT().
+		DeleteLink("id1", "id2").
+		Return(nil)
+
+	c := getCLIContext(t, Args{"name1", "name2"}, nil)
+	if err := command.Unlink(c); err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestEnvironmentUnlink_userInputErrors(t *testing.T) {
+	tc, ctrl := newTestCommand(t)
+	defer ctrl.Finish()
+	command := NewEnvironmentCommand(tc.Command())
+
+	contexts := map[string]*cli.Context{
+		"Missing SOURCE arg":      getCLIContext(t, Args{}, nil),
+		"Missing DESTINATION arg":     getCLIContext(t, Args{"name"}, nil),
+	}
+
+	for name, c := range contexts {
+		if err := command.Unlink(c); err == nil {
+			t.Fatalf("%s: error was nil!", name)
+		}
+	}
+}
+
+func TestEnvironmentUnlink_duplicateEnvironmentID(t *testing.T) {
```

Comment:
this probably isn't necessary since a user can never create a dup environment link